flink-core/src/main/java/org/apache/flink/configuration/Configuration.java

```diff
@@ -843,11 +843,14 @@ private float convertToFloat(Object o, float defaultValue) {
 		}
 		else if (o.getClass() == Double.class) {
 			double value = ((Double) o);
-			if (value <= Float.MAX_VALUE && value >= Float.MIN_VALUE) {
-				return (float) value;
-			} else {
+			if (value < -Float.MAX_VALUE
```

Comment:
I'm afraid in that case, 0 can not be treated right, so i add a test case to check.